File: app/src/main/java/ru/agpu/artikproject/background_work/widget/WidgetService.java
package ru.agpu.artikproject.background_work.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

import ru.agpu.artikproject.background_work.widget.providers.NewWidgetAdapter;

public class WidgetService extends RemoteViewsService {

    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new NewWidgetAdapter(getApplicationContext(), intent);
    }

}

File: app/src/main/java/ru/agpu/artikproject/background_work/widget/WidgetService.kt
package ru.agpu.artikproject.background_work.widget

import android.content.Intent
import android.widget.RemoteViewsService
import ru.agpu.artikproject.background_work.widget.providers.WidgetAdapter

class WidgetService: RemoteViewsService() {
    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
        return WidgetAdapter(applicationContext, intent)
    }
}